library/waku_thread/inter_thread_communication/requests/node_lifecycle_request.nim

```diff
@@ -49,19 +48,21 @@ proc destroyShared(self: ptr NodeLifecycleRequest) =
   deallocShared(self)
 
 proc createWaku(configJson: cstring): Future[Result[Waku, string]] {.async.} =
-  var conf: WakuNodeConf
+  var conf = defaultWakuNodeConf().valueOr:
+    return err("Failed creating node: " & error)
+
   var errorResp: string
 
   try:
-    if not parseConfig($configJson, conf, errorResp):
-      return err(errorResp)
-  except Exception:
-    return err("exception calling parseConfig: " & getCurrentExceptionMsg())
+    let jsonNode = parseJson($configJson)
 
-  # TODO: figure out how to extract default values from the config pragma
-  conf.nat = "any"
-  conf.maxConnections = 50.uint16
-  conf.maxMessageSize = default_values.DefaultMaxWakuMessageSizeStr
+    for confField, confValue in fieldPairs(conf):
+      if jsonNode.contains(confField):
+        # Make sure string doesn't contain the leading or trailing " character
+        var formattedString = ($jsonNode[confField]).strip(chars = {'\"'})
+        confValue = parseCmdArg(typeof(confValue), formattedString)
```

Comment:
Is this the line where the conf is overridden?

Reply:
Exactly! I'll add a comment, as it can be confusing